web/src/engine/osk/src/banner/suggestionBanner.ts

```diff
@@ -704,28 +704,39 @@ export class SuggestionBanner extends Banner {
       minWidth: 0,
     }
 
-    let totalWidth = 0;
-    let displayCount = 0;
-
-    let collapsedOptions: BannerSuggestion[] = [];
-
     for (let i=0; i<SuggestionBanner.SUGGESTION_LIMIT; i++) {
       const d = this.options[i];
 
       if(suggestions.length > i) {
         const suggestion = suggestions[i];
         d.update(suggestion, optionFormat);
-        if(d.collapsedWidth < d.expandedWidth) {
-          collapsedOptions.push(d);
-        }
-
-        totalWidth += d.collapsedWidth;
-        displayCount++;
       } else {
         d.update(null, optionFormat);
       }
     }
 
+    this.refreshLayout();
+  }
+
+  readonly refreshLayout = () => {
+    let collapsedOptions: BannerSuggestion[] = [];
+    let totalWidth = 0;
+
+    let displayCount = Math.min(this.currentSuggestions.length, 8);
```

Comment:
```suggestion
    let displayCount = Math.min(this.currentSuggestions.length, SuggestionBanner.SUGGESTION_LIMIT);
```

Reply:
Ack, should've checked this before the late-night merge.  Ah well.